src/accessible_name.ts

```diff
@@ -1,12 +1,32 @@
-/** This function is just an example to check the testing frameworks out
- * and will not be included in the actual implementation
- * @param {Node} textNode - A DOM node
- * @return {string} The accessible name for textNode
+import {rules} from './lib/rules';
+import {Context} from './lib/context';
+
+/**
+ * @param {HTMLElement} elem - The element whose accessible name will be calculated
+ * @param {Context} context - Additional information relevant to the name computation for elem
+ * @return {string} - The accessible name for elem
  */
-export function getTextNodeAccessibleName(textNode: Node): string {
-    let accessibleName: string = '';
-    if (textNode && textNode.nodeType === Node.TEXT_NODE && textNode.textContent) {
-        accessibleName = textNode.textContent;
+function _getAccessibleName(elem: HTMLElement, context: Context): string {
```

Comment:
don't use '_' as a prefix. Prefer naming your function with an appropriate name (computeTextAlternative might be good)

also getAccessibleName can keep an HTMLElement as a param, but this should probably take a Node.